sql/src/SqlIndex.sk

```diff
@@ -6,18 +6,37 @@ module alias P = SQLParser;
 
 module SKDB;
 
-class IndexProjKey(values: Array<(Int, ?CValue)>) extends SKStore.Key
-
-fun makeIndexProjKey(
-  v: RowValues,
+class IndexProjKey private (
+  row: RowValues,
   kinds: Array<(Int, P.IKind, P.Type)>,
```

Comment:
Other than the call to `compareRows`, only the `Int` component is used, so maybe a little more memory can be saved:
```suggestion
  kinds: Array<Int>,
```
Since these `kinds` values are always constructed with `P.IASC()`, only the first `Int` component of `kinds` is non-constant and used by `compareRows`, so a specialized version could be written.